src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -348,23 +348,23 @@ public BitArray And(BitArray value)
             ref int right = ref MemoryMarshal.GetArrayDataReference<int>(valueArray);
             if (Vector512.IsHardwareAccelerated && (uint)count >= Vector512<int>.Count)
             {
-                for (; i < (uint)count - (Vector512<int>.Count - 1u); i += (uint)Vector512<int>.Count)
+                for (; i < (uint)count - ((uint)Vector512<int>.Count - 1u); i += (uint)Vector512<int>.Count)
```

Comment:
why is this necessary?

Reply:
Mixing `int` and `uint` causes both to be extended to `long`, and 64bit arithmetic is slow in 32bit processes.

Reply:
The question is more "why use uint at all here"?

Just use the standard `int i = 0` instead and if then do `LoadUnsafe(ref left, (uint)i)` if you want to zero-extend the load.

Reply:
It seems https://github.com/dotnet/runtime/pull/33749 decided to use `uint i` instead of `int i` to avoid potential overflows. I don't see any potential overflow to necessitate `uint i` in the current version of BitArray.cs, though. Should I change all those `uint i` to `int i`?

Reply:
There shouldn't be any risk to overflow even in the previous implementation. The loop should never be incrementing past `Length` in the first place.

Using `int` tends to work better for the JIT and unlocks more overall optimizations. It will also improve readability here.